lib/seam/http_single_workspace.rb

```diff
@@ -18,9 +18,10 @@ def initialize(client: nil, api_key: nil, personal_access_token: nil, workspace_
         @endpoint = options[:endpoint]
         @auth_headers = options[:auth_headers]
         @defaults = Seam::DeepHashAccessor.new({"wait_for_action_attempt" => wait_for_action_attempt})
-
         @client = client || Seam::Http::Request.create_faraday_client(@endpoint, @auth_headers, faraday_options,
           faraday_retry_options)
+
+        initialize_routes(client: @client, defaults: @defaults)
```

Comment:
What does this do? Why is it defined as a private function but called here?

Reply:
`initialize_routes` is an internal method to set up route clients by making available `@client` and `@defaults` for them. `SingleWorkspace` can access private `initialize_routes` because `Routes` module gets `include`d in the class. While the method will exist on instances, being private it can only be called internally (inside the class/module methods) and not from outside code, unlike public methods like 'devices', 'access_codes' etc. which can be called by anyone using the instance.

Reply:
Why is this method needed though vs just putting the code inline? I think I must be missing something obvious.

Reply:
I was wondering why not to just do
```
      @client = client
      @defaults = defaults
``` 

But now I understand what this is doing. Basically this module only works if mixed into classes that define `@client` and `@defaults`. And this method is helping clarify that requirement.

I wonder if there is a more standard pattern to handle this.
I think it may be what this question is asking about https://stackoverflow.com/questions/12586051/initializing-instance-variables-in-mixins